pkg/graph/client.go

```diff
@@ -0,0 +1,114 @@
+package graph
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+
+	"github.com/the-maldridge/nbuild/pkg/types"
+)
+
+// NewAPIClient creates a new API client.
+func NewAPIClient(l hclog.Logger) *APIClient {
+	x := APIClient{
+		l: l.Named("client"),
```

Comment:
This client needs to have its own http.Client which has a timeout set.  Otherwise its possible for this to just hang forever.